src/renderer/components/cluster-manager/lens-views.ts

```diff
@@ -2,6 +2,8 @@ import { observable, when } from "mobx";
 import { ClusterId, clusterStore, getClusterFrameUrl } from "../../../common/cluster-store";
 import { getMatchedClusterId } from "../../navigation";
 import logger from "../../../main/logger";
+import AwaitLock from "await-lock";
```

Comment:
Can you explain in short what it does and why it's needed for lens-views? (it's not obvious and there is no description in PR)

Reply:
We were observing a race condition (the same as was also fixed with #2190) and this was my proposed solution. I still think it is relevant because we don't really want to have to think about mutable mutable access.

Reply:
Thanks, but I still can't fully get why it might happen? Which process modifies what which leads to race condition?

- `lens-views` is running in single process in main frame of `BrowserWindow`
- `initView` is called once on change `getMatchedClusterId()` and saves view to the map. where is race?

Reply:
I forget exactly what was causing it, but you could easily get it to happen if any of the async calls took a long time (for instance if we added a 5s delay). It would show multiple clusters at once.